Mathlib/Combinatorics/Additive/AP/Three/Defs.lean

```diff
@@ -103,40 +100,41 @@ theorem threeGPFree_singleton (a : α) : ThreeGPFree ({a} : Set α) :=
 
 @[to_additive ThreeAPFree.prod]
 theorem ThreeGPFree.prod {t : Set β} (hs : ThreeGPFree s) (ht : ThreeGPFree t) :
-    ThreeGPFree (s ×ˢ t) := fun _ _ _ ha hb hc h =>
+    ThreeGPFree (s ×ˢ t) := fun _ ha _ hb _ hc h ↦
   Prod.ext (hs ha.1 hb.1 hc.1 (Prod.ext_iff.1 h).1) (ht ha.2 hb.2 hc.2 (Prod.ext_iff.1 h).2)
 #align mul_salem_spencer.prod ThreeGPFree.prod
 #align add_salem_spencer.prod ThreeAPFree.prod
 
 @[to_additive]
 theorem threeGPFree_pi {ι : Type*} {α : ι → Type*} [∀ i, Monoid (α i)] {s : ∀ i, Set (α i)}
     (hs : ∀ i, ThreeGPFree (s i)) : ThreeGPFree ((univ : Set ι).pi s) :=
-  fun _ _ _ ha hb hc h =>
+  fun _ ha _ hb _ hc h ↦
   funext fun i => hs i (ha i trivial) (hb i trivial) (hc i trivial) <| congr_fun h i
 #align mul_salem_spencer_pi threeGPFree_pi
 #align add_salem_spencer_pi threeAPFree_pi
 
 end Monoid
 
 section CommMonoid
+variable [CommMonoid α] [CommMonoid β] {s A : Set α} {t B : Set β} {f : α → β} {a : α}
 
 variable [CommMonoid α] [CommMonoid β] {s : Set α} {a : α}
```

Comment:
Something went wrong with the variables here

Reply:
Fixed in #12546